src/Common/CustomAttributes/PreviewAttributeHelper.cs

```diff
@@ -33,16 +34,31 @@ internal class PreviewAttributeHelper
         public static void ProcessCustomAttributesAtRuntime(Type type, InvocationInfo invocationInfo, Action<string> writeOutput)
         {
             List<CmdletPreviewAttribute> attributes = new List<CmdletPreviewAttribute>(GetAllAttributesInType(type, invocationInfo));
+            StringBuilder sb = new StringBuilder().Clear();
+            Action<string> appendPreviewMessage = (string s) => sb.Append(s);
 
             if (attributes != null && attributes.Count > 0)
             {
                 foreach (CmdletPreviewAttribute attribute in attributes)
                 {
-                    attribute.PrintCustomAttributeInfo(writeOutput);
+                    attribute.PrintCustomAttributeInfo(appendPreviewMessage);
                 }
+                writeOutput(sb.ToString());
+
             }
         }
 
+        /// <summary>
+        /// Process CmdletExperimentation attribute in runtime
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="invocationInfo"></param>
+        /// <param name="writeOutput"></param>
+        public static bool ContainsPreviewAttribute(Type type, InvocationInfo invocationInfo)
+        {
+            return GetAllAttributesInType(type, invocationInfo).Count() > 0;
```

Comment:
Just want to double check that although the method is called `GetAllAttributesInType`, it only gets preview attribute and ignores other kinds of attributes.

Reply:
Yes, that's true.